GAE/src/org/waterforpeople/mapping/dataexport/RawDataSpreadsheetImporter.java

```diff
@@ -434,41 +445,41 @@ private Integer parseRepeatsForInstance(InstanceData instanceData,
         // 8. SurveyalTime - ignored duplicate
         // 9 - N. Questions
 
-        int instanceIdColumnIndex = metadataColumnHeaderIndex2.get(SURVEY_INSTANCE_COLUMN_KEY);
-        int repeatIterationColumnIndex = metadataColumnHeaderIndex2.get(REPEAT_COLUMN_KEY);
+        int identifierColumnIndex = metadataColumnHeaderIdx.get(DATAPOINT_IDENTIFIER_COLUMN_KEY);
+        int repeatIterationColumnIndex = metadataColumnHeaderIdx.get(REPEAT_COLUMN_KEY);
 
-        String SurveyInstanceId = instanceData.surveyInstanceDto.getKeyId().toString();
+        String dataPointIdentifier = instanceData.surveyInstanceDto.getSurveyedLocaleIdentifier();
         //Find first row matching this, or give up
         int rowIx = currentRowIndex;
         Row row;
         while (true) {
             row = repSheet.getRow(rowIx);
             if (row != null
-                    && row.getCell(instanceIdColumnIndex) != null
-                    && row.getCell(instanceIdColumnIndex).getStringCellValue().equals(SurveyInstanceId)) { //found!
+                    && row.getCell(identifierColumnIndex) != null
+                    && row.getCell(identifierColumnIndex).getStringCellValue().equals(dataPointIdentifier)) { //found!
                 break;
             } else {
-                if (isEmptyRow(row)) { // a row without any cells defined
-                    rowIx = 1; //end of sheet; start over
-                    }
-                rowIx++;
-                if (rowIx == currentRowIndex) { //back to where we started
-                    return rowIx; //not found; there were 0 iterations
+                if (rowIx > repSheet.getLastRowNum()) { //fell off the end
+                    rowIx = 1; //start over
                 }
             }
+            rowIx++;
+            if (rowIx == currentRowIndex) { //back to where we started
+                return rowIx; //not found; there were 0 iterations
+            }
         }
         //Found one row for the instance, read them all
 
         Map<Long, Map<Long, String>> responseMap = new HashMap<>();
 
         while (row != null
-                && row.getCell(instanceIdColumnIndex) != null
-                && row.getCell(instanceIdColumnIndex).getStringCellValue().equals(SurveyInstanceId) //TODO: use "identifier" as link instead to handle new data?
+                && row.getCell(identifierColumnIndex) != null
+                && row.getCell(identifierColumnIndex).getStringCellValue().equals(dataPointIdentifier)
                 && row.getCell(repeatIterationColumnIndex) != null
                 && row.getCell(repeatIterationColumnIndex).getCellType() == Cell.CELL_TYPE_NUMERIC) {
             Long rep = (long) row.getCell(repeatIterationColumnIndex).getNumericCellValue(); //might throw on huge number
-            //check rep for sanity
-            if (rep<1) { break;}
+            //check repeat no for sanity
+            if (rep < 1) { break;}
```

Comment:
Should we perhaps `continue` and just skip such a row but process the rest that are ok?